conans/util/files.py

```diff
@@ -88,7 +88,10 @@ def _detect_encoding(text):
                 return encodings[bom], len(bom)
             except UnicodeDecodeError:
                 continue
-    decoders = ["utf-8", "Windows-1252"]
+    if sys.platform == "win32":
+        decoders = ["utf-8", "mbcs", "Windows-1252"]
```

Comment:
This is a bit risky. At the very least I would try to:
- Add the encoder later, after ``Windows-1252``
- Add some test that reproduce the error and is solved by this change

Reply:
> This is a bit risky. At the very least I would try to:
> 
> * Add the encoder later, after `Windows-1252`
> * Add some test that reproduce the error and is solved by this change

I'm afarid that adding the encoder later may not work. `Windows-1252` is a single-byte code page and it covers almost every available value for a byte (with only 5 unused code points). For command line outputs, most of the short lines will succeed with `Windows-1252` decoder (which produces unreadable contents).

Testing this error requires altering the system locale settings. I'm still looking for a way to handle this.

Reply:
1252 is just one of the possible encodings, popular in certain European countries I assume? e.g. In my location (Asia) 1251 is much more wide-spread. I assume to fix a problem "world-wide" we would need to try a bunch of different encodings?

Reply:
> 1252 is just one of the possible encodings, popular in certain European countries I assume? e.g. In my location (Asia) 1251 is much more wide-spread. I assume to fix a problem "world-wide" we would need to try a bunch of different encodings?

I don't think it is possible to try those encodings. Most of mis-decoded contents are totally legal data, but only non-readable by human.

Reply:
yes, that's the problem. conan cannot know if an arbitrary file is cp1251, cp866 or koi8r. it can be decoded via any of the above, it will be just human-unreadable. unless conan aplies some heuristics (e.g. frequency of characters), there is no reliable way to deal with that code-page.
but if I read the issue correctly, it's child process launched by conan has a different text encoding from the original process, which caueses troubles. in that way I would prefer if child terminal just started with the "proper" (= same as parent) encoding.

Reply:
The way that conan handles normal Windows subprocess is to send them directly to the console without decoding them. If we do the same for bash subprocess (using PR #11306), at least the display will be correct (but using custom `output` still has problem).